configuration/src/main/java/io/airlift/configuration/ConfigurationInspector.java

```diff
@@ -52,6 +53,9 @@ private static String getValue(Method getter, Object instance, String defaultVal
             if (value == null) {
                 return "null";
             }
+            if (value instanceof Optional) {
+                return ((Optional<Object>) value).map(Object::toString).orElse("null");
```

Comment:
This is an incorrect cast. Cast to `Optional<?>` instead.

BTW isn't printing `null` a bit inappropriate for non-null optional values?
Wouldn't `----` (or something) be better?
(We could change `"null"` above for `value == null` case too)